Packs/EmailCommunication/Scripts/SendEmailReply/SendEmailReply.py

```diff
@@ -564,26 +572,36 @@ def get_incident_by_query(query):
     return incidents_details
 
 
-def get_unique_code():
+def get_unique_code(incident_id):
     """
-        Create an 8-digit unique random code that should be used to identify new created incidents.
-    Args: None
+        Create an 13-digit unique random code that should be used to identify new created incidents.
+    Args:
+        incident_id:
+
     Returns:
-        8-digit code returned as a string
+        13-digit code returned as a string
     """
+    demisto.debug(f'Generating a unique code for incident {incident_id}')
+    tried_codes = set()
+    incident_id_padded = incident_id[-3:].rjust(3, "0")  # Take padded last 3 digits of incident ID.
     while True:
-        code = f'{random.randrange(1, 10 ** 8):08}'
-        query = f'emailgeneratedcode: {code}'
-        incidents_details = get_incident_by_query(query)
-        if incidents_details is None or len(incidents_details) == 0:
-            return code
+        # The random code is 13 digits long and is created by concatenating the last 3 digits of the incident ID and epoch.
+        code = f'{incident_id_padded}{int(time.time()):010d}'
```

Comment:
This resolution is for seconds. Don't you want to be more sure, and use millis? e.g.
```suggestion
        code = f'{incident_id_padded}{int(1000*time.time()):013d}'
```